expose more pulp parameters

timogoebel opened the pull request:
This PR exposed more pulp parameters so users have more choices what they want.
This is in preparation to manage pulp also for a pulp master by this module.

ekohl commented on manifests/init.pp:
```diff
@@ -26,6 +26,20 @@
+#
+# $pulp_num_workers::                   Number of Pulp workers to use.
+#
+# $pulp_proxy_password::                Proxy password for authentication
```
I'd prefer this to be under username.

timogoebel commented:
@ekohl: I finally found time to revisit this.

* rebased
* moved the parameter password parameter under the username parameter
* added a parameter for the ca_cert

ekohl commented on manifests/init.pp:
```diff
@@ -20,6 +20,22 @@
+#
+# $pulp_puppet_wsgi_processes::         Number of WSGI processes to spawn for the puppet webapp
+#
+# $pulp_ca_cert::                       CA certificate for pulp
```
I feel this is a bit brief. The puppet-pulp description is: Absolute path to PEM encoded CA certificate file, used by Pulp to validate the identity of the broker using SSL.

ekohl commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
+  Optional[String] $pulp_proxy_url = $foreman_proxy_content::params::pulp_proxy_url,
+  Optional[String] $pulp_proxy_username = $foreman_proxy_content::params::pulp_proxy_username,
+  Optional[Integer] $pulp_puppet_wsgi_processes = $foreman_proxy_content::params::pulp_puppet_wsgi_processes,
+  Optional[String] $pulp_ca_cert = $foreman_proxy_content::params::pulp_ca_cert,
```
This can be ```Optional[Stdlib::Absolutepath]```

ekohl commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
+  Optional[Integer] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
+  Optional[Integer] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
+  Optional[String] $pulp_proxy_password = $foreman_proxy_content::params::pulp_proxy_password,
+  Optional[Integer] $pulp_proxy_port = $foreman_proxy_content::params::pulp_proxy_port,
```
This can limit to ```[0, 65535]```

ekohl commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
   Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
```
Should this be at least 1?

ekohl commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
+  Optional[Integer] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
```
Should this be at least 1?

ekohl commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
+  Optional[Integer] $pulp_proxy_port = $foreman_proxy_content::params::pulp_proxy_port,
+  Optional[String] $pulp_proxy_url = $foreman_proxy_content::params::pulp_proxy_url,
+  Optional[String] $pulp_proxy_username = $foreman_proxy_content::params::pulp_proxy_username,
+  Optional[Integer] $pulp_puppet_wsgi_processes = $foreman_proxy_content::params::pulp_puppet_wsgi_processes,
```
Should this be at least 1?

timogoebel commented:
@ekohl: Addressed your comments, thanks.

ekohl commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
   Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer[1,]] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
```
I'm surprised the trailing comma works but I think it's cleaner without in this case.

timogoebel commented:
@ekohl: Done.

stbenjam commented on manifests/init.pp:
```diff
@@ -52,6 +68,14 @@
   String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
   Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+  Optional[Integer[1]] $pulp_max_tasks_per_child = $foreman_proxy_content::params::pulp_max_tasks_per_child,
+  Optional[Integer[1]] $pulp_num_workers = $foreman_proxy_content::params::pulp_num_workers,
```
Shouldn't these get passed to pulp class?

timogoebel commented:
@stbenjam: Thanks, added the missing params.

ekohl commented:
Thanks!
